### PR TITLE
Split build jobs into seperate jobs

### PR DESCRIPTION
**Pull request by Hoxbro**
Split up the job Conda job into three separate jobs. One for Conda itself, one for NPM, and one for the CDN upload.

For these 3 I have added an upload artifact which should make it easy to inspect the build if needed. 

The NPM I have used `npm pack` to be able to save the tarball as an artifact, and then I use that tarball to upload with `npm upload`. 

Also took the time to clean up the build file.

### CODE REVIEW

File: scripts/build_pyodide_wheels.py
Comment (Hoxbro) [inline scripts/build_pyodide_wheels.py:72]: contourpy is now available on pyodide

**Comment by Hoxbro** [comment]
Need to double-check if the wheel files are correctly uploaded to the cdn.

File: .github/workflows/build.yaml
Comment (philippjfr) [inline .github/workflows/build.yaml:75]: I would like this to at least be dependent on either the conda or the pip build passing.

File: .github/workflows/build.yaml
Comment (philippjfr) [inline .github/workflows/build.yaml:132]: Same with this.

File: .github/workflows/build.yaml
Comment (Hoxbro) [reply .github/workflows/build.yaml:75]: Was thinking about having deploy/publish step separated for the build, which needs approval and requires all the building steps to pass, so we can ensure that everything is ready before publishing. 

Inspiration for this could be https://timheuer.com/blog/add-approval-workflow-to-github-actions/

Comment (maximlt) [reply .github/workflows/build.yaml:75]: > we can ensure that everything is ready before publishing

I'm still unsure why we'd need that. Isn't that the purpose of the dev releases?

Comment (Hoxbro) [reply .github/workflows/build.yaml:75]: Things can still go wrong. The dev and final releases are fundamentally independent; e.g., a new package can be released between the dev and final releases.

This will also give extra security if a wrong tag has been pushed, you forgot to pull before tagging locally, or you forgot to merge PRs.

Comment (maximlt) [reply .github/workflows/build.yaml:75]: Let's bring this up in a HoloViz meeting or a more general channel. We need to ensure that the repos are structured and behave consistently.

**Review by philippjfr — APPROVED** [review]

**Pull request merged**
